src/components/invitation/ItemOptionModal.tsx

```diff
@@ -77,13 +83,27 @@ const ItemOptionModal: FunctionComponent<ItemOptionModalProps> = ({
       {}
     ) as OptionDictOptionData;
     let haveAllMandatoryOption = true;
-    listify(dishOption, (_, value) => ({ ...value })).forEach((menuOption) => {
+    listify(dishOption, (key, value) => ({ optionId: key, ...value })).forEach((menuOption) => {
       if (!menuOption.isMandatory || !haveAllMandatoryOption) {
         return;
       }
-      if (!get(currentDishOption, menuOption.id.toString())) {
-        haveAllMandatoryOption = false;
+      const selectedOptionItems = currentDishOptionMap[menuOption.optionId]?.value;
+      if (Array.isArray(selectedOptionItems)) { // choice
+        const { minQuantity, maxQuantity } = menuOption;
+        if (
+          selectedOptionItems.length >= minQuantity 
+          && selectedOptionItems.length <= maxQuantity
+          && selectedOptionItems.every(({ optionItemId }) => !!get(menuOption.items, optionItemId.toString()))
+        ) {
+          return;
+        }
+      } else if (
+        !!selectedOptionItems 
+        && get(menuOption.items, selectedOptionItems.optionItemId?.toString())
+      ) { // mandatory
```

Comment:
you can try to unify `selectedOptionItems` to array to eliminate the need to check for this

Reply:
Fixed